addons/sale_timesheet/models/product.py

```diff
@@ -76,7 +76,10 @@ def _compute_product_tooltip(self):
     @api.onchange('type', 'service_type', 'service_policy')
     def _onchange_service_fields(self):
         for record in self:
-            if record.type == 'service' and record.service_type == 'timesheet' and \
+            default_uom = self.env['ir.default']._get_model_defaults('product.template').get('uom_id')
+            if default_uom:
+                record.uom_id = self.env['uom.uom'].browse(default_uom)
+            elif record.type == 'service' and record.service_type == 'timesheet' and \
```

Comment:
We can use the default value choose by the user but I think we should check if that default uom is a unit or a unit of time. If not, then we cannot use that UoM for timesheet service.